src/repl.coffee

```diff
@@ -7,7 +7,8 @@ CoffeeScript = require './coffeescript'
 
 replDefaults =
   prompt: 'coffee> ',
-  historyFile: path.join process.env.HOME, '.coffee_history' if process.env.HOME
+  historyFile: do (p = process.env.XDG_CACHE_HOME or process.env.HOME) ->
+    path.join p, '.coffee_history' if p
   historyMaxInputSize: 10240
```

Comment:
Fair point on the variable name.

Current functionality is not to write a history file if `$HOME` is unset, which from my perspective is absolutely correct. We don't know what kind of weird environment we have if `$HOME` isn't set, and I know from playing around with other tools like `git` that they prefer not to guess about this sort of thing. One can imagine a user with no write access to `$PWD`; what a pain for them if `coffee` simply refuses to run!